flexinput/src/main/java/com/lytefast/flexinput/adapters/PhotoCursorAdapter.kt

```diff
@@ -136,12 +165,25 @@ class PhotoCursorAdapter(private val contentResolver: ContentResolver,
     fun bind(photo: Photo?) {
       this.photo = photo
 
-      val thumbnailUri = photo?.let {
-        setSelected(selectionCoordinator.isSelected(photo, adapterPosition), false)
-        it.getThumbnailUri(contentResolver)
-      }
+      if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+        imageView.hierarchy.setPlaceholderImage(placeholderDrawable, ScalingUtils.ScaleType.CENTER)
+
+        loadingThumbnailJob = GlobalScope.launch(context = Dispatchers.Main) {
+          val thumbnailDrawable = BitmapDrawable(imageView.resources, getThumbnailAsync())
```

Comment:
couple things here:

1. we have hidden the "asynchronous" part deep inside of this line. can we make a val just for the bitmap? and then use it in the drawable. Then that `getThumbnailAsync` really stands out, which is good for understandability.
2. what happens when the thumbnail fails to load? that thumbnail returns null right? have we tested that flow by forcing a null return after some delay?

Reply:
1) Good call
2) I actually encountered this in testing, a thumbnail throwing an exception so I wrapped the `loadThumbnail` in an exception and log the failures

Reply:
Re: 2 -- yes that makes sense but what happens? Does the image just stay empty? Are we sure?